code/controllers/subsystems/machines.dm

```diff
@@ -72,25 +72,26 @@ if(current_step == this_step || (check_resumed && !resumed)) {\
 			propagate_network(PC,PC.powernet)
 
 datum/controller/subsystem/machines/proc/setup_atmos_machinery(list/machines)
-	set background=1
 
 	if(!Master.current_runlevel)//So it only does it at roundstart
 		report_progress("Initializing atmos machinery")
+	var/list/atmosmachines = list()
 	for(var/obj/machinery/atmospherics/A in machines)
 		A.atmos_init()
+		atmosmachines += A
 		CHECK_TICK
 
-	for(var/obj/machinery/atmospherics/unary/U in machines)
-		if(istype(U, /obj/machinery/atmospherics/unary/vent_pump))
-			var/obj/machinery/atmospherics/unary/vent_pump/T = U
-			T.broadcast_status()
-		else if(istype(U, /obj/machinery/atmospherics/unary/vent_scrubber))
-			var/obj/machinery/atmospherics/unary/vent_scrubber/T = U
+	for(var/U in atmosmachines)
+		if(istype(U, /obj/machinery/atmospherics/unary/vent_pump) || istype(U, /obj/machinery/atmospherics/unary/vent_scrubber))
+			var/obj/machinery/atmospherics/unary/T = U
 			T.broadcast_status()
 		CHECK_TICK
 	if(!Master.current_runlevel)//So it only does it at roundstart
 		report_progress("Initializing pipe networks")
-	for(var/obj/machinery/atmospherics/machine in machines)
+	for(var/m in atmosmachines)
```

Comment:
What the heck is this block?

Put it back to the way it was before. m can never be null here